File: helpdesk-pro/frontend/src/app/core/services/saved-view.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models';

export interface SavedView {
  id: string;
  name: string;
  filterJson: string;
  createdAt: string;
}

@Injectable({ providedIn: 'root' })
export class SavedViewService {
  private base = `${environment.apiUrl}/saved-views`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<SavedView[]> {
    return this.http.get<ApiResponse<SavedView[]>>(this.base).pipe(map(r => r.data));
  }

  create(name: string, filterJson: string): Observable<SavedView> {
    return this.http.post<ApiResponse<SavedView>>(this.base, { name, filterJson }).pipe(map(r => r.data));
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}

File: helpdesk-mobile/lib/core/services/saved_view_service.dart
import '../api/api_client.dart';
import '../api/api_endpoints.dart';
import '../models/saved_view_model.dart';

class SavedViewService {
  final ApiClient _api;
  SavedViewService(this._api);

  Future<List<SavedViewModel>> getAll() async {
    final resp = await _api.get(ApiEndpoints.savedViews);
    final data = resp.data['data'];
    final list = data is List ? data : (data['content'] ?? []);
    return List<SavedViewModel>.from(list.map((e) => SavedViewModel.fromJson(e)));
  }

  Future<SavedViewModel> create(Map<String, dynamic> data) async {
    final resp = await _api.post(ApiEndpoints.savedViews, data: data);
    return SavedViewModel.fromJson(resp.data['data']);
  }

  Future<SavedViewModel> update(String id, Map<String, dynamic> data) async {
    final resp = await _api.put(ApiEndpoints.savedView(id), data: data);
    return SavedViewModel.fromJson(resp.data['data']);
  }

  Future<void> delete(String id) async {
    await _api.delete(ApiEndpoints.savedView(id));
  }
}
